sqlx-core/src/postgres/types/chrono.rs

```diff
@@ -80,6 +121,52 @@ impl<Tz: TimeZone> Type<Postgres> for Vec<DateTime<Tz>> {
     }
 }
 
+impl Encode<'_, Postgres> for PgTimeTz {
+    fn encode_by_ref(&self, buf: &mut PgArgumentBuffer) -> IsNull {
+        let _ = <NaiveTime as Encode<'_, Postgres>>::encode(self.time, buf); // IsNull::No
+        <i32 as Encode<'_, Postgres>>::encode(self.offset.utc_minus_local(), buf)
+    }
+
+    fn size_hint(&self) -> usize {
+        mem::size_of::<i64>() + mem::size_of::<i32>()
+    }
+}
+
+impl<'r> Decode<'r, Postgres> for PgTimeTz {
+    fn decode(value: PgValueRef<'r>) -> Result<Self, BoxDynError> {
+        match value.format() {
+            PgValueFormat::Binary => {
+                let mut buf = Cursor::new(value.as_bytes()?);
+
+                // TIME is encoded as the microseconds since midnight
+                let us = buf.read_i64::<BigEndian>()?;
+                let time = NaiveTime::from_hms(0, 0, 0) + Duration::microseconds(us);
+
+                // OFFSET is encoded as seconds from UTC
+                let seconds = buf.read_i32::<BigEndian>()?;
+                Ok(PgTimeTz::new(time, FixedOffset::west(seconds)))
+            }
+
+            PgValueFormat::Text => {
+                // Please dig into PostgreSQL source code and see if we can
+                // implement parsing for this. Chrono doesn't really support
+                // parsing of times only with a timezone in a format PostgreSQL
+                // gives us, so it needs to be something custom.
+                //
+                // See `timetz_in` in `adt/date.c` and `ParseDateTime` in
+                // `dt_common.c`.
+
+                let error = io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    "Reading a `TIMETZ` value in text format is not supported.",
+                );
+
+                Err(Box::new(error))
```

Comment:
If you're returning `Box<dyn Error>` you can literally just return the error string via the `impl<'a, '_> From<&'_ str> for Box<dyn Error + Send + Sync + 'a>`:

```rust
Err("Reading a `TIMETZ` value in text format is not supported.".into())
```

Reply:
:+1: